aspnetcore/tutorials/first-mongo-app.md

```diff
@@ -166,79 +167,108 @@ The database is ready. You can start creating the ASP.NET Core web API.
   * `collection` represents the collection name.
-  * `T` represents the CLR object type stored in the collection.
+  * `TDocument` represents the CLR object type stored in the collection.
 
-`GetCollection<T>(collection)` returns a `MongoCollection` object representing the collection. In this tutorial, the following methods are invoked on the collection:
+`GetCollection<TDocument>(collection)` returns a [MongoCollection](https://api.mongodb.com/csharp/current/html/T_MongoDB_Driver_MongoCollection.htm) object representing the collection. In this tutorial, the following methods are invoked on the collection:
 
-* `Find<T>` &ndash; Returns all documents in the collection matching the provided search criteria.
-* `InsertOne` &ndash; Inserts the provided object as a new document in the collection.
-* `ReplaceOne` &ndash; Replaces the single document matching the provided search criteria with the provided object.
-* `DeleteOne` &ndash; Deletes a single document matching the provided search criteria.
+* [DeleteOne](https://api.mongodb.com/csharp/current/html/M_MongoDB_Driver_IMongoCollection_1_DeleteOne.htm) &ndash; Deletes a single document matching the provided search criteria.
+* [Find\<TDocument>](https://api.mongodb.com/csharp/current/html/M_MongoDB_Driver_IMongoCollectionExtensions_Find__1_1.htm) &ndash; Returns all documents in the collection matching the provided search criteria.
+* [InsertOne](https://api.mongodb.com/csharp/current/html/M_MongoDB_Driver_IMongoCollection_1_InsertOne.htm) &ndash; Inserts the provided object as a new document in the collection.
+* [ReplaceOne](https://api.mongodb.com/csharp/current/html/M_MongoDB_Driver_IMongoCollection_1_ReplaceOne.htm) &ndash; Replaces the single document matching the provided search criteria with the provided object.
 
 ## Add a controller
 
-1. Add a `BooksController` class to the *Controllers* directory with the following code:
+Add a `BooksController` class to the *Controllers* directory with the following code:
 
-    [!code-csharp[](first-mongo-app/sample/BooksApi/Controllers/BooksController.cs)]
+[!code-csharp[](first-mongo-app/sample/BooksApi/Controllers/BooksController.cs)]
 
-    The preceding web API controller:
+The preceding web API controller:
+
+* Uses the `BookService` class to perform CRUD operations.
+* Contains action methods to support GET, POST, PUT, and DELETE HTTP requests.
+* Calls <xref:System.Web.Http.ApiController.CreatedAtRoute*>, in the `Create` action method, to return an [HTTP 201](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) response. 201 is the standard response for an HTTP POST method that creates a new resource on the server. `CreatedAtRoute` also adds a `Location` header to the response. The `Location` header specifies the URI of the newly created book.
```

Comment:
* Essential phrase (dropping commas).
* Sentence starting with "201" ... I put in "Status code 201" ... something like that.

```suggestion
* Calls <xref:System.Web.Http.ApiController.CreatedAtRoute*> in the `Create` action method to return an [HTTP 201](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) response. Status code 201 is the standard response for an HTTP POST method that creates a new resource on the server. `CreatedAtRoute` also adds a `Location` header to the response. The `Location` header specifies the URI of the newly created book.
```